plugin/pkg/admission/securitycontext/scdeny/admission.go

```diff
@@ -53,42 +60,65 @@ func (p *plugin) Admit(a admission.Attributes) (err error) {
 		return apierrors.NewBadRequest("Resource was marked with kind Pod but was unable to be converted")
 	}
 
-	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.SupplementalGroups != nil {
-		return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("SecurityContext.SupplementalGroups is forbidden"))
+	if err = checkPodSecurityContext(pod.Spec.SecurityContext); err != nil {
+		return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, err)
 	}
-	if pod.Spec.SecurityContext != nil {
-		if pod.Spec.SecurityContext.SELinuxOptions != nil {
-			return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("pod.Spec.SecurityContext.SELinuxOptions is forbidden"))
-		}
-		if pod.Spec.SecurityContext.RunAsUser != nil {
-			return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("pod.Spec.SecurityContext.RunAsUser is forbidden"))
+
+	for _, container := range pod.Spec.InitContainers {
+		if err = checkSecurityContext(container.SecurityContext); err != nil {
+			return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, err)
 		}
 	}
 
-	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.FSGroup != nil {
-		return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("SecurityContext.FSGroup is forbidden"))
+	for _, container := range pod.Spec.Containers {
+		if err = checkSecurityContext(container.SecurityContext); err != nil {
+			return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, err)
+		}
 	}
 
-	for _, v := range pod.Spec.InitContainers {
-		if v.SecurityContext != nil {
-			if v.SecurityContext.SELinuxOptions != nil {
-				return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("SecurityContext.SELinuxOptions is forbidden"))
-			}
-			if v.SecurityContext.RunAsUser != nil {
-				return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("SecurityContext.RunAsUser is forbidden"))
-			}
+	return
+}
+
+func checkPodSecurityContext(podContext *api.PodSecurityContext) error {
```

Comment:
Because I was talking about [extracting method](https://refactoring.com/catalog/extractMethod.html) I meant that you wouldn't change the original code but mostly copy&paste it. Now I'm asking you to preserve the original code instead of using rewritten version of it.

Example, how it should be:
```golang
func checkSecurityContext(context *api.SecurityContext) error {
    if context == nil {
        return nil
    }
    if context.SELinuxOptions != nil {
        return fmt.Errorf("SecurityContext.SELinuxOptions is forbidden")
    }
    if context.RunAsUser != nil {
        return fmt.Errorf("SecurityContext.RunAsUser is forbidden")
    }
}
```